rust/src/table/state.rs

```diff
@@ -290,6 +302,14 @@ impl DeltaTableState {
             self.min_writer_version = new_state.min_writer_version;
         }
 
+        for feature in new_state.reader_features.iter() {
+            self.reader_features.insert(feature.to_owned());
+        }
+
+        for feature in new_state.writer_features.iter() {
+            self.writer_features.insert(feature.to_owned());
+        }
```

Comment:
It's unclear to me why we should take the union, and not just adopt the latest set of features. This would mean you could never drop features, right?